packages/material-ui/src/Select/SelectInput.js

```diff
@@ -15,17 +15,14 @@ import useForkRef from '../utils/useForkRef';
 import useControlled from '../utils/useControlled';
 import selectClasses, { getSelectUtilitiyClasses } from './selectClasses';
 
+// TODO: enable overrides for the icon and nativeInput slots via the components prop
 export const overridesResolver = (props, styles) => {
   const { styleProps } = props;
   return deepmerge(
     {
       ...styles.select,
-      ...styles[styleProps.variant],
-      [`& .${selectClasses.icon}`]: {
-        ...styles.icon,
-        ...(styleProps.variant && styles[`icon${capitalize(styleProps.variant)}`]),
-        ...(styleProps.open && styles.iconOpen),
-      },
+      [`&.${selectClasses.selectMenu}`]: styles.selectMenu,
+      [`&.${selectClasses[styleProps.variant]}`]: styles[styleProps.variant],
     },
     styles.root || {},
```

Comment:
https://github.com/mui-org/material-ui/blob/d72862cf60b2ea314b5d9e3ed163a0f605e2eef2/packages/material-ui/src/NativeSelect/NativeSelectInput.js#L25

Reply:
https://github.com/mui-org/material-ui/blob/d72862cf60b2ea314b5d9e3ed163a0f605e2eef2/packages/material-ui/src/Select/SelectInput.js#L356-L358

Reply:
Referencing #11646 where we plan to simply this

Reply:
And #25313

Reply:
Should we do this in a separate PR?

Reply:
Probably, yes